controllers/cruisecontroloperation_controller.go

```diff
@@ -0,0 +1,445 @@
+		Namespace: operation.GetNamespace(),
+	}, nil
+}
+
+func updateResult(res *scale.Result, operation *banzaiv1alpha1.CruiseControlOperation, isAfterExecution bool) error {
+	if res == nil || operation.GetCurrentTask() == nil {
+		return nil
+	}
+	task := operation.GetCurrentTask()
+
+	if (task.State == banzaiv1beta1.CruiseControlTaskCompleted || task.State == banzaiv1beta1.CruiseControlTaskCompletedWithError) && task.Finished == nil {
+		task.Finished = &v1.Time{Time: time.Now()}
+	}
+
+	// Add the failed task into the status.failedTasks slice only when the update is happened after executing task
+	if isAfterExecution && task.Finished != nil && task.State == banzaiv1beta1.CruiseControlTaskCompletedWithError {
+		if len(operation.Status.FailedTasks) >= defaultFailedTasksHistoryMaxLength {
+			operation.Status.FailedTasks = append(operation.Status.FailedTasks[1:], *task)
+		} else {
+			operation.Status.FailedTasks = append(operation.Status.FailedTasks, *task)
+		}
+		operation.Status.NumberOfRetries += 1
+		operation.Status.ErrorPolicy = banzaiv1alpha1.ErrorPolicyRetry
+		task.Finished = nil
+		task.State = ""
+		task.Started = nil
+	}
+
+	task.ID = res.TaskID
+	if task.Started == nil {
+		//Sat, 27 Aug 2022 12:22:21 GMT
+		startTime, err := time.Parse(time.RFC1123, res.StartedAt)
+		if err != nil {
+			return errors.WrapIff(err, "could not parse user task start time from Cruise Control API")
+		}
+		task.Started = &v1.Time{Time: startTime}
+	}
+	// We only have summary after execution
+	if isAfterExecution {
+		task.Summary = parseSummary(res.Result)
+		task.ErrorMessage = res.Err
+	}
+	// This is necessary because of a bug in CC (when a user task resulted in error later when get usertasks status from CC the corresponding task state is Active and not CompletedWithError...)
+	//if task.State != banzaiv1beta1.CruiseControlTaskCompletedWithError && task.State != banzaiv1beta1.CruiseControlTaskCompleted {
+	task.State = res.State
+	//}
+	return nil
+}
+
+// updateActiveTasks updates the state of the CruiseControlOperation from the CruiseControlTasksAndStates instance by getting their
+// status from Cruise Control.
+func (r *CruiseControlOperationReconciler) updateCurrentTasks(ctx context.Context, ccOperations []*banzaiv1alpha1.CruiseControlOperation) error {
+	log := logr.FromContextOrDiscard(ctx)
+	tasks, err := r.Scaler.GetUserTasks()
+	if err != nil {
+		return errors.WrapIff(err, "could not get user tasks from Cruise Control API")
+	}
+
+	var ccOperationsCopy []*banzaiv1alpha1.CruiseControlOperation
+	for _, ccOperation := range ccOperations {
+		copy := ccOperation.DeepCopy()
+		ccOperationsCopy = append(ccOperationsCopy, copy)
+	}
+
+	taskResultsByID := make(map[string]*scale.Result, len(tasks))
+	for _, task := range tasks {
+		taskResultsByID[task.TaskID] = task
+	}
+	for i := range ccOperations {
+		ccOperation := ccOperations[i]
+		log.Info("taskres current", "res", taskResultsByID[ccOperation.GetCurrentTaskID()])
```

Comment:
Is this meant to be `"current tasks"`? Also, I don't think the key "res" is clear enough when it is in the log message where we don't have context of the source code

Reply:
This is only for debugging for myself. I will remove it.